backend/server/routers/user.py

```diff
@@ -147,3 +147,46 @@ def update_degree_length(numYears: int, token: str = DUMMY_TOKEN):
     else:
         user['planner']['years'] = user['planner']['years'][:diff]
     set_user(token, user, True)
+
+@router.put("/setProgram") 
+def setProgram(token: str = DUMMY_TOKEN, degree: dict):
+    user = get_user(token);
+    user['user']['degree']['programCode'] = degree['programCode']
+    set_user(token, user, True)
+
+@router.put("/addSpecialisation") 
+def addSpecialisation(token: str = DUMMY_TOKEN, specialisation: str):
+    user = get_user(token);
+    user['user']['degree']['specs'].append(specialisation)
+    set_user(token, user, True)
+
+@router.put("/removeSpecialisation") 
+def removeSpecialisation(token: str = DUMMY_TOKEN, specialisation: str):
+    user = get_user(token);
+    specs = user['user']['degree']['specs']
+    if specialisation in specs:
+        user['user']['degree']['specs'] = specs.remove(specialisation)
```

Comment:
`.remove` on lists is in-place and returns None so, this is sufficient as

```suggestion
        specs.remove(specialisation)
```